### PR TITLE
feat(modal): add ability to inject default modal options

### LINKED ISSUE #5719: Setting default options for Modal?
**Description:**

How could I set modal default options, in other words, override `modalConfigDefaults`?

There is a [way](https://valor-software.com/ngx-bootstrap/#/tooltip#config-defaults) to do it for Tooltip component. Could we achieve a such with Modal?
Thanks.

### PR DESCRIPTION
**Pull request by dkosasih**
This enable us to declare global ModalOption on the module level so that we do not have to pass in the ModalOption everytime we call ModalService.show().
`{provide: MODAL_CONFIG_DEFAULT_OVERRIDE, useValue: {<Default Options>}}`

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated tests.

closes #5719

### CODE REVIEW

**Comment by daniloff200** [comment]
hey, @dkosasih , I see some conflicts here. Could you please fix them?

File: src/spec/modal.default-setting.spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+class TestModalComponent { }
+
+@NgModule({
+    declarations: [TestModalComponent],
+    providers: [{ provide: MODAL_CONFIG_DEFAULT_OVERRIDE, useValue: { ignoreBackdropClick: true, keyboard: false } }],
+    entryComponents: [TestModalComponent]
+})
+export class TestModule { }
+
+fdescribe('Modal service', () => {
```
Comment (daniloff200) [inline src/spec/modal.default-setting.spec.ts]: please, remove **f**

**Review by daniloff200 — CHANGES_REQUESTED** [review]

**Comment by dkosasih** [comment]
@daniloff200, I have made the necessary changes. Please have a look.

**Comment by daniloff200** [comment]
Well, more than 2 years passed, and looks like, we ready to check that out
Gonna do that this week

**Review by daniloff200 — APPROVED** [review]

**Comment by ghost** [comment]
I try to use this feature in `AppComponent`:
```
import { MODAL_CONFIG_DEFAULT_OVERRIDE } from 'ngx-bootstrap/modal/modal-options.class';
@Component({
  ...
  providers: [
    ...
    {
      provide: MODAL_CONFIG_DEFAULT_OVERRIDE,
      useValue: { keyboard: false }
    }
    ...
  ]
  ...
})
...
```
Building ends to an error:
```
ERROR in ./src/app/app.component.ts
Module not found: Error: Can't resolve 'ngx-bootstrap/modal/modal-options.class' in <path>\src\app
```
I am using Angular 10.0.14.
Is the cause that Angular does not support deep imports now?
Should `MODAL_CONFIG_DEFAULT_OVERRIDE` be exported from `ngx-bootstrap/modal/public_api.d.ts`?

Thank you for any help!

**Comment by daniloff200** [comment]
you can try `ngx-bootstrap/modal` @wickedsp1d3r

**Comment by dkosasih** [comment]
If this is due to Angular not supporting deep import. We need to expose the `MODAL_CONFIG_DEFAULT_OVERRIDE` through public_api. It's not exported at the moment.

**Comment by ghost** [comment]
Yes, I get an error if I export it from `ngx-bootstrap/modal`:
```
Module '"../../node_modules/ngx-bootstrap/modal/ngx-bootstrap-modal"' has no exported member 'MODAL_CONFIG_DEFAULT_OVERRIDE'
```

**Comment by dkosasih** [comment]
yes. That is due to the lack of exposure to public_api. I have open another PR here: https://github.com/valor-software/ngx-bootstrap/pull/5923

**Comment by eduardoluizm** [comment]
@dkosasih  I have the same error with 6.1.0 version and Angular 10.

"Module not found: Error: Can't resolve 'ngx-bootstrap/modal/modal-options.class'"

How can I fix it?

**Pull request merged**